index.js

```diff
@@ -11,11 +12,51 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
+server.get('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(payment){
+        res.status(200).send(payment);
+    }
+    else{
+        res.status(404).send({message: 'Wrong id'});
+    }
+});
+
+server.post('/payments', (req, res) => {
+    const payment = {...req.body};
+    Object.assign(payment, {
+        id: uuidv4(),
+        status: 'Pending'
+    });
+    payments.push(payment);
+    res.status(200).send(payment);
+});
 
+server.put('/payments/:id', (req, res) => {
+    const payment = payments.find( payment => payment.id === req.params.id);
+    if(payment){
```

Comment:
this is good now. 
what we can do even much better is decoupling these 2 validation checks for  `payment` and `req.body.description` . currently the latter one is nested under the first one. Can you see how we can write `if()` conditions with `return`  keyword and without `else` part ? 
that will increase the code readability.